src/main/java/Duke.java

```diff
@@ -1,10 +1,272 @@
+                    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                } else if (command.startsWith("mark")) {
+                    int index = Integer.parseInt(command.split(" ")[1]) - 1;
+                    if (index >= 0 && index < tasks.size()) {
+                        tasks.get(index).markDone();
+                        System.out.println("Nice! I've marked this task as done:\n  " + tasks.get(index));
+                    } else {
+                        System.out.println("Invalid task index.");
+                    }
+                } else if (command.startsWith("unmark")) {
+                    int index = Integer.parseInt(command.split(" ")[1]) - 1;
+                    if (index >= 0 && index < tasks.size()) {
+                        tasks.get(index).markNotDone();
+                        System.out.println("OK, I've marked this task as not done yet:\n  " + tasks.get(index));
+                    } else {
+                        System.out.println("Invalid task index.");
+                    }
+                }  else if (command.startsWith("delete")) {
+                    int index = Integer.parseInt(command.split(" ")[1]) - 1;
+                    if (index >= 0 && index < tasks.size()) {
+                        Task removedTask = tasks.remove(index);
+                        System.out.println("Noted. I've removed this task:\n  " + removedTask);
+                        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                    } else {
+                        System.out.println("Invalid task index.");
+                    }
+                } else {
+                    throw new DukeException("I'm sorry, but I don't know what that means :-(");
+                }
+                saveTasksToFile(tasks);
+            }
+            catch (DukeException e) {
+                System.out.println("☹ OOPS!!! " + e.getMessage());
+            }
+
+            System.out.println("____________________________________________________________");
+        }
+
+
+        scanner.close();
+    }
+    private static String formatDate(LocalDate date) {
+        return date.format(DateTimeFormatter.ofPattern("MMM dd yyyy"));
+    }
+    private static void loadTasksFromFile(ArrayList<Task> tasks) {
+        try {
+            File file = new File(FILE_PATH);
+            if (file.exists()) {
+                Scanner fileScanner = new Scanner(file);
+                while (fileScanner.hasNextLine()) {
+                    String line = fileScanner.nextLine();
+                    Task task = createTaskFromLine(line);
+                    if (task != null) {
+                        tasks.add(task);
+                    }
+                }
+                fileScanner.close();
+            }
+        } catch (FileNotFoundException e) {
+            // Handle file not found exception
+            System.out.println("File not found: " + FILE_PATH);
```

Comment:
I like that the handling of this exception not only tells that user that the file cannot be found but also returns the filepath